### PR TITLE
LPTIM INIT function

### PR DESCRIPTION
**Pull request by alejandrogonzalvo**
#  ⌛ LPTIM INIT function ⏳ 
This PR introduces the LPTIM init function inside the ADC init function, which overloads the HAL init.

## Before
```cpp
Pin::start();
MX_DMA_Init();
ADC::start();
MX_LPTIM1_Init();
MX_LPTIM2_Init();
MX_LPTIM3_Init();
```

## After
```cpp
Pin::start();
MX_DMA_Init();
ADC::start();
```

## Testing:
It has been tested with the same testcase of the ADC, deleting the HAL inits and checking that it still works correctly

### CODE REVIEW

**Comment by ricardochust** [comment]
this pull request is marked merge with main. Make it merge with HALAL or development

**Comment by alejandrogonzalvo** [comment]
> this pull request is marked to merge with main. Make it merge with HALAL or development

Done

**Comment by ricardochust** [comment]
is modifying language.settings necessary by any means? if not don t modify it

**Comment by ricardochust** [comment]
last thing i see is that you didn t mention if you tested it and how. Could you write it on a comment or update the pull request?

**Comment by alejandrogonzalvo** [comment]
> last thing i see is that you didn t mention if you tested it and how. Could you write it on a comment or update the pull request?

Done

File: Inc/HALAL/Models/LowPowerTimer/LowPowerTimer.hpp [hunk 1]
```diff
@@ -6,16 +6,22 @@
  */
 
 #pragma once
-#ifdef HAL_LPTIM_MODULE_ENABLED
 
+//#ifdef HAL_LPTIM_MODULE_ENABLED
```
Comment (PabloGonzalezSegarra) [inline Inc/HALAL/Models/LowPowerTimer/LowPowerTimer.hpp:10]: Esto porque esta comentado?

Comment (alejandrogonzalvo) [reply Inc/HALAL/Models/LowPowerTimer/LowPowerTimer.hpp:10]: bien visto, lo descomento

**Review by StefanCostea — APPROVED** [review]

**Comment by ricardochust** [comment]
> is modifying language.settings necessary by any means? if not don t modify it

-

**Review by ricardochust — APPROVED** [review]

**Review by PabloGonzalezSegarra — APPROVED** [review]

**Pull request merged**